activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -145,5 +150,9 @@ def can_be_bound?(column_name, value)
         handler_for(value).is_a?(BasicObjectHandler) &&
         !table.associated_with?(column_name)
     end
+
+    def polymorphic_array?(value)
+      value.is_a?(Array) && value.map(&:class).map { |e| e.try(:base_class) }.uniq.size > 1
```

Comment:
Take a look at the implementation of `associated_with?`, where we look up
whether the reflection object is present.

On Thu, Jan 7, 2016 at 5:56 PM Philippe Huibonhoa <notifications@github.com>
wrote:

> In activerecord/lib/active_record/relation/predicate_builder.rb
> <https://github.com/rails/rails/pull/22365#discussion_r49147757>:
>
> > @@ -145,5 +150,9 @@ def can_be_bound?(column_name, value)
> >          handler_for(value).is_a?(BasicObjectHandler) &&
> >          !table.associated_with?(column_name)
> >      end
> > +
> > +    def polymorphic_array?(value)
> > +      value.is_a?(Array) && value.map(&:class).map { |e| e.try(:base_class) }.uniq.size > 1
>
> @sgrif <https://github.com/sgrif> Sorry, for my unfamiliarity, but at
> this point value would be an array of AR objects, not necessarily a
> relation, so I don't think I would have access to polymorphic? If that's
> the case, I'm going to try your suggestion of moving the switch to a
> factory method on AssociationQueryValue and move this there with an early
> exit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/22365/files#r49147757>.
>